backend/apiServer/controller/repo.go

```diff
@@ -88,3 +90,87 @@ func (repo RepoController) NewRepoFromURI(w http.ResponseWriter, r *http.Request
 
 	return
 }
+
+
+/**
+* @api {Post} /repo/:id Parse the repository assosiated with id.
+* @apiName Parse repository.
+* @apiGroup Repository
+* @apiPermission none
+*
+* @apiParam {String} Id Id of submitted git repository.
+*
+* @apiParamExample {url} Parse repository:
+* 	/repo/:id
+*
+* @apiSuccessExample {json} Success-Response:
+* 	HTTP/1.1 200 OK
+*	{
+*	    "functions": [
+*	        {
+*	            "file": "main.cpp",
+*	            "function_names": [
+*	                {
+*	                    "name": "int main()"
+*	                }
+*	            ]
+*	        }
+*	    ]
+*	}
+*
+* @apiErrorExample {json} Post invalid id.
+*	HTTP/1.1 400 Bad Request
+*	{
+*		Invalid parameters
+*	}
+*
+* @apiErrorExample {json} Internal error.
+*	HTTP/1.1 500 Internal Server Error
+*	{
+*		Internal Server Error
+*	}
+*
+ */
+
+// ParseSimpleFunc parse a repository for functions of a certain project in repos directory.
+func (repo RepoController) ParseSimpleFunc(w http.ResponseWriter, r *http.Request) {
+	http.Header.Add(w.Header(), "content-type", "application/json")
+
+	if r.Method == "POST" {
+		id := strings.TrimPrefix(r.URL.Path, "/repo/")
+
+		// Validate that the project exist in DB.
+		exstRepo, err := model.RepoModel{}.GetRepoByID(id)
+
+		if err != nil {
+			http.Error(w, "Invalid parameters", http.StatusBadRequest)
+			log.Println("Could not find repo in db: ", err.Error())
+			return
+		}
+
+		// List all files in the repository directory.
+		files, err := exstRepo.GetRepoFile(model.RepoPath + "/" + exstRepo.ID.Hex())
```

Comment:
exstRepo contains id.Hex() given to it as a parameter and model.Repopath is here taken from model into controller and sendt back into model. could just have getrepo use models local variables and exstRepo that it works on (no param needed)